src/common/low_precision_transformations/src/recurrent_cell.cpp

```diff
@@ -91,8 +62,132 @@ RecurrentCellTransformation::RecurrentCellTransformation(const Params& params) :
     this->register_matcher(m, callback);
 }
 
+namespace {
+
+std::shared_ptr<ov::opset1::FakeQuantize> find_fake_quantize_upper(const std::shared_ptr<Node>& parent) {
+    if (is_type<ov::opset1::FakeQuantize>(parent)) {
+        return as_type_ptr<ov::opset1::FakeQuantize>(parent);
+    }
+
+    if (!NetworkHelper::isPrecisionPreserved(parent)) {
+        return nullptr;
+    }
+
+    return find_fake_quantize_upper(parent->get_input_node_shared_ptr(0));
+}
+
+template <class Operation>
+std::string name() {
+    return Operation::get_type_info_static().name;
+}
+
+bool isSupportedForPerChannelQuantization(const std::shared_ptr<Node>& node) {
+    static const std::unordered_set<std::string> precisionPreservedOps = {
+        { name<opset1::DepthToSpace>() },
+        { name<opset1::Interpolate>() },
+        { name<opset1::MaxPool>() },
+        { name<opset1::ReduceMax>() },
+        { name<opset1::ReduceMin>() },
+        { name<opset1::Relu>() },
+        { name<opset2::BatchToSpace>() },
+        { name<opset1::Broadcast>() },
+        { name<opset3::Broadcast>() },
+        { name<opset1::Pad>() },
+        { name<opset12::Pad>() },
+        { name<opset1::Reshape>() },
+        { name<opset1::Squeeze>() },
+        { name<opset2::SpaceToBatch>() },
+        { name<opset1::Split>() },
+        { name<opset1::StridedSlice>() },
+        { name<opset1::ShuffleChannels>() },
+        { name<opset1::Transpose>() },
+        { name<opset1::Unsqueeze>() },
+        { name<opset1::VariadicSplit>() }
+    };
+
+    return precisionPreservedOps.find(node->get_type_name()) != precisionPreservedOps.end();
+}
+
+std::vector<std::pair<size_t, element::Type>> get_quantized_inputs(std::shared_ptr<ov::Node> lstm) {
+    return is_type<ov::opset5::LSTMSequence>(lstm) ?
```

Comment:
if the node is not GRU and LSTM, we need to throw exception here I suppose

Reply:
why not? was added